blockchain/ethereum/internal/chain.go

```diff
@@ -38,13 +39,19 @@ type ChainBackend struct {
 }
 
 // NewFunder initializes and returns an instance of ethereum funder.
-func (cb *ChainBackend) NewFunder(assetAddr pwallet.Address) pchannel.Funder {
-	return pethchannel.NewETHFunder(*cb.Cb, pethwallet.AsEthAddr(assetAddr))
+func (cb *ChainBackend) NewFunder(assetAddr pwallet.Address, onChainAddr pwallet.Address) pchannel.Funder {
+	asset := pethwallet.AsWalletAddr(pethwallet.AsEthAddr(assetAddr))
+	onChainAcc := accounts.Account{Address: pethwallet.AsEthAddr(onChainAddr)}
+	onChainAccs := map[pethchannel.Asset]accounts.Account{*asset: onChainAcc}
+	depositors := map[pethchannel.Asset]pethchannel.Depositor{*asset: new(pethchannel.ETHDepositor)}
+	return pethchannel.NewFunder(*cb.Cb, onChainAccs, depositors)
 }
 
 // NewAdjudicator initializes and returns an instance of ethereum adjudicator.
 func (cb *ChainBackend) NewAdjudicator(adjAddr, receiverAddr pwallet.Address) pchannel.Adjudicator {
-	return pethchannel.NewAdjudicator(*cb.Cb, pethwallet.AsEthAddr(adjAddr), pethwallet.AsEthAddr(receiverAddr))
+	receiverAcc := accounts.Account{Address: pethwallet.AsEthAddr(receiverAddr)}
+	return pethchannel.NewAdjudicator(*cb.Cb, pethwallet.AsEthAddr(adjAddr),
+		pethwallet.AsEthAddr(receiverAddr), receiverAcc)
```

Comment:
As is "onChainAddr" everywhere else, including the interface definition for the same function in perun.go, the same term can be used here as well for consistency,
```suggestion
	onChainAcc := accounts.Account{Address: pethwallet.AsEthAddr(onChainAddr)}
	return pethchannel.NewAdjudicator(*cb.Cb, pethwallet.AsEthAddr(adjAddr),
		pethwallet.AsEthAddr(onChainAddr), onChainAcc)
```

Reply:
[Renamed](https://github.com/hyperledger-labs/perun-node/pull/114/commits/7070f0127dead417d64fc18a13603559ced67452).